kafka/src/main/java/io/streamzi/cloudevents/kafka/util/KafkaHeaderUtil.java

```diff
@@ -2,56 +2,64 @@
 import org.apache.kafka.clients.consumer.ConsumerRecord;
-import org.apache.kafka.common.header.Header;
 import org.apache.kafka.common.header.Headers;
 import org.apache.kafka.common.header.internals.RecordHeader;
 import org.apache.kafka.common.header.internals.RecordHeaders;
+import org.apache.kafka.common.serialization.Serde;
 
 import java.net.URI;
-import java.net.URISyntaxException;
-import java.time.Instant;
-import java.time.ZoneOffset;
 import java.time.ZonedDateTime;
 
+import static io.streamzi.cloudevents.impl.CloudEventImpl.CLOUD_EVENTS_VERSION_KEY;
+import static io.streamzi.cloudevents.impl.CloudEventImpl.CONTENT_TYPE_KEY;
+import static io.streamzi.cloudevents.impl.CloudEventImpl.EVENT_ID_KEY;
+import static io.streamzi.cloudevents.impl.CloudEventImpl.EVENT_TIME_KEY;
+import static io.streamzi.cloudevents.impl.CloudEventImpl.EVENT_TYPE_KEY;
+import static io.streamzi.cloudevents.impl.CloudEventImpl.EVENT_TYPE_VERSION_KEY;
+import static io.streamzi.cloudevents.impl.CloudEventImpl.SCHEMA_URL_KEY;
+import static io.streamzi.cloudevents.impl.CloudEventImpl.SOURCE_KEY;
+import static java.time.format.DateTimeFormatter.ISO_ZONED_DATE_TIME;
+
 /**
  * Utility methods for working with CloudEvents and Kafka.
  *
  */
-public class KafkaHeaderUtil {
+public final class KafkaHeaderUtil {
 
-    private static final String EVENT_TYPE = "EventType";
-    private static final String CLOUDEVENTS_VERSION = "CloudEventsVersion";
-    private static final String SOURCE = "Source";
-    private static final String EVENT_ID = "EventID";
-    private static final String EVENT_TYPE_VERSION = "EventTypeVersion";
-    private static final String SCHEMA_URL = "SchemaURL";
-    private static final String CONTENT_TYPE = "ContentType";
+    private KafkaHeaderUtil() {
+        // no-op
+    }
 
     /**
      * Create Kafka Headers from a CloudEvent
      * @param ce Event to extract the headers from
      * @return Headers that can be used to construct a ProducerRecord
      */
-    public static Iterable<Header> extractHeaders(final CloudEvent<?> ce) {
+    public static Headers extractHeaders(final CloudEvent<?> ce) {
 
-        RecordHeaders headers = new RecordHeaders();
+        final RecordHeaders headers = new RecordHeaders();
 
-        headers.add(new RecordHeader(EVENT_TYPE, ce.getEventType().getBytes()));
-        headers.add(new RecordHeader(CLOUDEVENTS_VERSION, ce.getCloudEventsVersion().getBytes()));
-        headers.add(new RecordHeader(SOURCE, ce.getSource().toString().getBytes()));
-        headers.add(new RecordHeader(EVENT_ID, ce.getEventID().getBytes()));
+        headers.add(new RecordHeader(EVENT_TYPE_KEY, ((Serde) CafdiSerdes.serdeFrom(ce.getEventType().getClass())).serializer().serialize(null, ce.getEventType())));
+        headers.add(new RecordHeader(CLOUD_EVENTS_VERSION_KEY, ((Serde) CafdiSerdes.serdeFrom(ce.getCloudEventsVersion().getClass())).serializer().serialize(null, ce.getCloudEventsVersion())));
```

Comment:
I think we could just use `String.class`, since we _know_ and _control_ the type's class

Reply:
Is using Kafka serialisation into the Headers necessary?